cclient/response/were_addresses_spent_from.c

```diff
@@ -17,16 +17,18 @@ void were_addresses_spent_from_res_free(were_addresses_spent_from_res_t* res) {
   utarray_free(res);
 }
 
-bool were_addresses_spent_from_res_at(were_addresses_spent_from_res_t* in,
-                                      int index) {
-  if (utarray_len(in) > index) {
-    bool* p = NULL;
-    p = (bool*)utarray_eltptr(in, index);
-    if (p) {
-      return *p;
+retcode_t were_addresses_spent_from_res_at(were_addresses_spent_from_res_t* res,
+                                           int index, bool** states) {
```

Comment:
why use `bool** states` instead of `bool* states`?

Reply:
Since I hope even though the user uses an uninitialized bool pointer as the argument of 'were_addresses_spent_from_res_at()' I can promise the 'were_addresses_spent_from_res_at()'
will return NULL

Reply:
Should I change it to 'bool* states' ?